apps/antalmanac/src/lib/enrollmentHistory.ts

```diff
@@ -0,0 +1,158 @@
+
+        const res = (await queryGraphQL<EnrollmentHistoryGraphQLResponse>(queryString))?.data?.enrollmentHistory;
+
+        if (res?.length) {
+            const parsedEnrollmentHistory = DepartmentEnrollmentHistory.parseEnrollmentHistoryResponse(res);
+            DepartmentEnrollmentHistory.sortEnrollmentHistory(parsedEnrollmentHistory);
+
+            // For now, just return the enrollment history of the most recent quarter
+            // instead of the entire array of enrollment histories
+            const latestEnrollmentHistory = parsedEnrollmentHistory[0];
+            return latestEnrollmentHistory;
+        }
+
+        return null;
+    }
+
+    /**
+     * This function parses enrollment history data from PeterPortal so that
+     * we can pass the data into a recharts graph. For each element in the given
+     * array, merge the dates, totalEnrolledHistory, maxCapacityHistory,
+     * and waitlistHistory arrays into one array that contains the enrollment data
+     * for each day.
+     *
+     * @param res an array of enrollment histories from PeterPortal
+     * @returns an array of enrollment histories that we can use for the graph
+     */
+    static parseEnrollmentHistoryResponse(res: EnrollmentHistoryGraphQL[]): EnrollmentHistory[] {
+        const parsedEnrollmentHistory: EnrollmentHistory[] = [];
+
+        for (const enrollmentHistory of res) {
+            const enrollmentDays: EnrollmentHistoryDay[] = [];
+
+            for (const [i, date] of enrollmentHistory.dates.entries()) {
+                const d = new Date(date);
+                const formattedDate = `${d.getMonth() + 1}/${d.getDate() + 1}/${d.getFullYear()}`;
+
+                enrollmentDays.push({
+                    date: formattedDate,
+                    totalEnrolled: Number(enrollmentHistory.totalEnrolledHistory[i]),
+                    maxCapacity: Number(enrollmentHistory.maxCapacityHistory[i]),
+                    waitlist:
+                        enrollmentHistory.waitlistHistory[i] === 'n/a'
+                            ? null
+                            : Number(enrollmentHistory.waitlistHistory[i]),
+                });
+            }
+
+            parsedEnrollmentHistory.push({
+                year: enrollmentHistory.year,
+                quarter: enrollmentHistory.quarter,
+                department: enrollmentHistory.department,
+                courseNumber: enrollmentHistory.courseNumber,
+                days: enrollmentDays,
+                instructors: enrollmentHistory.instructors,
+            });
+        }
+
+        return parsedEnrollmentHistory;
+    }
+
+    /**
+     * This function sorts the given array of enrollment histories so that
+     * the most recent quarters are in the beginning of the array.
+     *
+     * @param enrollmentHistory an array where each element represents the enrollment
+     * history of a course section during one quarter
+     */
```

Comment:
How I'd write comments. Definitely important since AA is a writing workshop 🤓 
```suggestion
    /**
     * Sort the given array of enrollment histories so that
     * the most recent quarters are in the beginning of the array.
     *
     * @param enrollmentHistory Array where each element represents the enrollment
     * history of a course section during one quarter.
     */
```